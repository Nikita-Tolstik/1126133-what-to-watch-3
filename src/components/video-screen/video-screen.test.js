import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';
import NameSpace from '../../reducer/name-space.js';
import VideoScreen from './video-screen.jsx';
import history from '../../history.js';

const mockStore = configureStore([]);

const children = <div className="children-component" />;

const mockFilm = {
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald2`,
  posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundColor: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  description: `Fantastic Beasts: The Crimes of Grindelwald`,
  rating: 5,
  scoresCount: 7,
  director: `Wes Andreson`,
  stars: [`Bill Murray`, `Edward Norton`, `Jude Law`],
  runTime: 55,
  genre: `Thriller`,
  released: 44,
  isFavorite: true,
  videoLink: `https://upload.wikimedia.org/`,
  videoPreview: `https://upload.wikimedia.org/`,
};

const Setting = {
  currentTime: 150,
  duration: 800,
};

it(`Render VideoScreen - isPlaying = true`, () => {

  const store = mockStore({
    [NameSpace.DATA]: {
      isError: false,
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <VideoScreen
            film={mockFilm}
            currentTime={Setting.currentTime}
            duration={Setting.duration}
            isPlaying={true}
            onFullScreenClick={() => {}}
            onPlayButtonClick={() => {}}
          >
            {children}
          </VideoScreen>
        </Provider>
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();

});

it(`Render VideoScreen - isPlaying = false`, () => {

  const store = mockStore({
    [NameSpace.DATA]: {
      isError: true,
    }
  });

  const tree = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <VideoScreen
            film={mockFilm}
            currentTime={Setting.currentTime}
            duration={Setting.duration}
            isPlaying={false}
            onFullScreenClick={() => {}}
            onPlayButtonClick={() => {}}
          >
            {children}
          </VideoScreen>
        </Provider>
      </Router>
  ).toJSON();

  expect(tree).toMatchSnapshot();

});
