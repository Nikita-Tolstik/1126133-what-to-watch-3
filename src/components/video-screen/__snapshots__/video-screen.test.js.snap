// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render VideoScreen - isPlaying = false 1`] = `
<div
  className="player"
>
  <div
    className="children-component"
  />
  <a
    className="player__exit"
    href="/films/1"
    onClick={[Function]}
  >
    Exit
  </a>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="19"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "19%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        10:50
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Fantastic Beasts: The Crimes of Grindelwald2
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
  <div
    className="error"
  >
    <p
      className="error__message"
    >
      Internal Server Error. Please, refresh page.
    </p>
    <button
      className="error__button"
      href="#"
      onClick={[Function]}
    >
      Refresh page
    </button>
  </div>
</div>
`;

exports[`Render VideoScreen - isPlaying = true 1`] = `
<div
  className="player"
>
  <div
    className="children-component"
  />
  <a
    className="player__exit"
    href="/films/1"
    onClick={[Function]}
  >
    Exit
  </a>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="19"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "19%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        10:50
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          height="21"
          viewBox="0 0 14 21"
          width="14"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Fantastic Beasts: The Crimes of Grindelwald2
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
